galsim/compound.py

```diff
@@ -645,6 +744,33 @@ def __str__(self):
         s += ')'
         return s
 
+    def _prepareDraw(self):
+        self._orig_obj._prepareDraw()
+        self.SBProfile = galsim._galsim.SBAutoConvolve(self._orig_obj.SBProfile, self._real_space,
+                                                       self._gsparams)
+
+    def shoot(self, n_photons, rng=None):
+        """Shoot photons into a PhotonArray.
+
+        @param n_photons    The number of photons to use for photon shooting.
+        @param rng          If provided, a random number generator to use for photon shooting,
+                            which may be any kind of BaseDeviate object.  If `rng` is None, one
+                            will be automatically created, using the time as a seed.
+                            [default: None]
+        @returns PhotonArray.
+        """
+        # Setup the rng if not provided one.
+        if rng is None:
```

Comment:
> why we shouldn't make it so this whole snippet equivalent to `ud = galsim.UniformDeviate(rng)`

+1 for this suggestion.  But the way we do the wrapping currently, it's not a trivial change.  If you think you know what to do to implement this, go for it.  Otherwise, I can give it a go.